02-peer-protocol.md

```diff
@@ -239,6 +243,8 @@ The sending node:
     - MUST set it to a defined type representing the type it wants.
     - MUST use the smallest bitmap possible to represent the channel type.
     - SHOULD NOT set it to a type containing a feature which was not negotiated.
+    - if `announce_channel` is `true` (not `0`):
+      - MUST NOT send `channel_type` with the `option_scid_alias` bit set.
```

Comment:
This needs some kind of feature gate - prior to this PR sending a channel_ready after the channel has advanced state` is not allowed (and, at least, we would barf on it).

Reply:
Maybe we just only allow this with the SCID alias channel type bit? There's not much reason to do it otherwise.

Reply:
Yea, that seems reasonable, the gating here should be on the feature bit, not the channel type.